File: 2017/day_5/quiz_2.rb
input = File.readlines('input.txt').map{ |line| line.strip.to_i }
range = (0..input.size - 1)
cursor = 0
steps = 0

while range.include?(cursor)
  move = input[cursor]
  if move >= 3
    input[cursor] -= 1
  else
    input[cursor] += 1
  end

  cursor += move

  steps+=1
end

p steps

File: 2024/day_1/quiz_2.py
from collections import Counter

def read_input(filename):
  try:
    with open(filename, 'r') as file:
      return file.readlines()
  except FileNotFoundError:
    print(f'File {filename} not found')
  except Exception as e:
    print(f'An error occurred: {e}')

if __name__ == '__main__':
  input_data = read_input('input.txt')
  left_list = []
  right_list = []
  for line in input_data:
    left, right = map(int, line.strip().split())
    left_list.append(left)
    right_list.append(right)

  occurrencies = Counter(right_list)
  sum_counts = sum(num * occurrencies[num] for num in left_list)
  print(sum_counts)
